haystack/preview/components/audio/whisper_remote.py

```diff
@@ -0,0 +1,144 @@
+from typing import List, Optional, Dict, Any, Union, BinaryIO, Literal, get_args, Sequence
+
+import os
+import json
+import logging
+from pathlib import Path
+from dataclasses import dataclass
+
+import requests
+from tenacity import retry, wait_exponential, retry_if_not_result
+
+from haystack.preview import component, Document
+from haystack.errors import OpenAIError, OpenAIRateLimitError
+
+
+logger = logging.getLogger(__name__)
+
+
+OPENAI_TIMEOUT = float(os.environ.get("HAYSTACK_OPENAI_TIMEOUT_SEC", 600))
+
+
+WhisperRemoteModel = Literal["whisper-1"]
+
+
+@component
+class RemoteWhisperTranscriber:
+    """
+    Transcribes audio files using OpenAI's Whisper using OpenAI API. Requires an API key. See the
+    [OpenAI blog post](https://beta.openai.com/docs/api-reference/whisper for more details.
+    You can get one by signing up for an [OpenAI account](https://beta.openai.com/).
+
+    For the supported audio formats, languages, and other parameters, see the
+    [Whisper API documentation](https://platform.openai.com/docs/guides/speech-to-text)
+    """
+
+    @dataclass
+    class Output:
+        documents: List[Document]
+
+    def __init__(self, api_key: str, model_name_or_path: WhisperRemoteModel = "whisper-1"):
+        """
+        Transcribes a list of audio files into a list of Documents.
+
+        :param api_key: OpenAI API key.
+        :param model_name_or_path: Name of the model to use. It now accepts only `whisper-1`.
+        """
+        if model_name_or_path not in get_args(WhisperRemoteModel):
+            raise ValueError(
+                f"Model name not recognized. Choose one among: " f"{', '.join(get_args(WhisperRemoteModel))}."
+            )
+
+        self.api_key = api_key
```

Comment:
Would it make sense to check if `api_key` is an empty string in here? 🤔

Reply:
Ok makes sense. You have to really do `RemoteWhisperTranscriber(api_key=None)` to manage that, but thinking about it, if you use `RemoteWhisperTranscriber(api_key=os.environ.get("API_KEY"))` that's likely. Adding the check!